test/queue-test.js

```diff
@@ -648,17 +640,31 @@ describe('Queue', (it) => {
         redisScanCount: 50
       });
 
-      queue.process(async () => {});
 
       // Choose a big number for numbers of jobs created, because otherwise the
       // set will be encoded as an intset and SSCAN will ignore the COUNT param.
       // https://redis.io/commands/scan#the-count-option
+      const createJobPromises = [];
       for (let i = 0; i < 10000; i++) {
-        await queue.createJob({foo: 'bar'}).save();
+        createJobPromises.push(queue.createJob({foo: 'bar'}).save());
       }
+      Promise.all(createJobPromises);
```

Comment:
you should `await` this, otherwise it'll process asynchronously (and you should probably define the `process` before creating the jobs)